pages/flashbar/common.ts

```diff
@@ -15,22 +15,29 @@ export const i18nStrings = {
   inProgressIconAriaLabel: 'In progress',
 };
 
-export function generateItem(
-  type: FlashbarProps.Type,
-  dismiss: (index: string) => void,
+export function generateItem({
+  type,
+  dismiss,
+  id = pseudoRandom().toString(),
   hasHeader = false,
-  initial = false
-): FlashbarProps.MessageDefinition {
-  const randomKey = generateUniqueId('key_');
+  initial = false,
+}: {
+  type: FlashbarProps.Type;
+  dismiss: (index: string) => void;
+  id?: string;
+  hasHeader?: boolean;
+  initial?: boolean;
+}): FlashbarProps.MessageDefinition {
   return {
     type,
-    id: randomKey,
+    id,
     dismissible: true,
     dismissLabel: 'Dismiss',
-    onDismiss: () => dismiss(randomKey),
+    onDismiss: () => dismiss(id),
     buttonText: 'Do Action',
+    onButtonClick: () => null,
```

Comment:
Adding `onButtonClick` gets rid of a console warning about passing `buttonText` but not this other property.

Reply:
`buttonText`/`onButtonClick` API is deprecated, as you can see in API docs

If you want to fix it, replace `buttonText` with `action: <Button>Do action</Button>`